### PR TITLE
[SPARK-29862][SQL] CREATE (OR REPLACE) ... VIEW should look up catalog/table like v2 commands

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Add CreateViewStatement and make CREARE VIEW  go through the same catalog/table resolution framework of v2 commands.

### Why are the changes needed?
It's important to make all the commands have the same table resolution behavior, to avoid confusing end-users. e.g.
```
USE my_catalog
DESC v // success and describe the view v from my_catalog
CREATE VIEW v AS SELECT 1 // report view not found as there is no view v in the session catalog
```
### Does this PR introduce any user-facing change?
Yes. When running CREATE VIEW ...  Spark fails the command if the current catalog is set to a v2 catalog, or the view name specified a v2 catalog.

### How was this patch tested?
unit tests

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 1]
```diff
@@ -3227,6 +3227,60 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
...
+    val persistedView: Int = 1
+    val globalTempView: Int = 2
+    val localTempView: Int = 3
+    val viewType = if (ctx.TEMPORARY == null) {
+      persistedView
+    } else if (ctx.GLOBAL != null) {
+      globalTempView
+    } else {
+      localTempView
+    }
```
Comment (huaxingao) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: I can't access ```PersistedView/GlobalTempView/LocalTempView``` in this class. I use 1/2/3 and map them back to ```PersistedView/GlobalTempView/LocalTempView``` in ```ResolveSessionCatalog```

Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: Can we move PersistedView etc.?

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala (hunk 1 above)
Comment (huaxingao) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: Yes. We can move these to catalyst. What is a good place to put these? ```org.apache.spark.sql.catalyst.analysis.view.scala```? ```org.apache.spark.sql.catalyst.catalog.interface.scala```?

Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: I think view.scala is better. cc @cloud-fan

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: +1 for view.scala

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala [hunk 2]
```diff
@@ -1669,6 +1669,105 @@ class DDLParserSuite extends AnalysisTest {
...
+    val expected2 = CreateViewStatement(
+      Seq("a", "b", "c"),
+      Seq("col1" -> None, "col3" -> Some("hello")),
+      Some("BLABLA"),
+      Map("prop1Key" -> "prop1Val"),
+      Some("SELECT * FROM tab1"),
+      parsePlan("SELECT * FROM tab1"),
+      false,
+      true,
+      PersistedView)
```
Comment (viirya) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala]: these tests are all for PersistedView, can we have few for other view types?

Comment (huaxingao) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala]: Sure. Updated.

**Review by viirya — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by huaxingao** [comment]
Thanks! @viirya @cloud-fan

**Pull request closed without merging**